### ISSUE TITLE
Retrieve preferences by section

### ISSUE THREAD

**Issue by rvignesh89** [EliotBerriot/django-dynamic-preferences#125]
Firstly I think sections in preferences is a neat idea. It helps me to logically split preferences in my app. However using restframework I'm not able to retrieve a bunch of preferences by section as currently we can only retrieve preferences individually or all of them.

@EliotBerriot do you think it is possible to have the list endpoint to be able to retrieve by section like below?

```bash
/api/preferences/ # retrieve all preferences
/api/preferences/section/ # retrieve all under section
/api/preferences/section__name/ # retrieve specific preference
```

I don't think it would be difficult to do as well 😃

**Comment by EliotBerriot** [comment]
I think this would be a great addition. The URL scheme you suggested may not be implementable in a backward-compatible manner, though, because if sections and single preferences share the same endpoint, we could end up with collisions.

However, there is a way to implement in a backward-compatible (and future-proof) manner, by using filters.

Basically, instead of using `/api/preferences/section/ ` to list all preferences of a given section, you would do: `/api/preferences/?section=my_section`. This scheme is also more powerful, because it means you could combine filters if we add more in the future: `/api/preferences/?section=my_section&type=boolean`.

[It's supported out of the box](http://www.django-rest-framework.org/api-guide/filtering/#djangofilterbackend) by django-rest-framework :)

**Comment by rvignesh89** [comment]
Ah I missed that. I just tried it with `django-filters` and it works as you mentioned. Thanks!

**Comment by EliotBerriot** [comment]
If you don't mind, I'll reopen your issue, as I think it's something reasonable to have in the package ;)

**Comment by rvignesh89** [comment]
I understand. It would be a nice feature to have. Do you mind if I try to implement this?

**Comment by EliotBerriot** [comment]
Not at all, treat yourself!

**Comment by EliotBerriot** [comment]
Fixed in #154
